### ISSUE TITLE
Normalization layer Exception: Tries to access Cudnn which isn't available (Theano-Backend)

### ISSUE THREAD

**Issue by Y4register** [fchollet/keras#4821]
I have a model which works well. When trying to add a batch normalization layer (uncomment its code line) it throws the exception mentioned below.

Cudnn isn't installed and isn't available on my device (not available to my NVIDIA Card). I don't know why exactly it's veing used...

```
top_model.add(Convolution2D(512, 3, 3, activation='relu', name='conv5_3'))
# top_model.add(BatchNormalization())
top_model.add(Flatten())
top_model.add(Dense(1024, activation='relu'))
```

>   File "exp.py", line 8, in <module>
>     main_function()
>   File "/directory/fine_tune_cnn.py", line 149, in main_function
>     top_model.add(BatchNormalization())
>   File "/directory/ve/lib/python2.7/site-packages/keras/models.py", line 331, in add
>     output_tensor = layer(self.outputs[0])
>   File "/directory/ve/lib/python2.7/site-packages/keras/engine/topology.py", line 569, in __call__
>     self.add_inbound_node(inbound_layers, node_indices, tensor_indices)
>   File "/directory/ve/lib/python2.7/site-packages/keras/engine/topology.py", line 632, in add_inbound_node
>     Node.create_node(self, inbound_layers, node_indices, tensor_indices)
>   File "/directory/ve/lib/python2.7/site-packages/keras/engine/topology.py", line 164, in create_node
>     output_tensors = to_list(outbound_layer.call(input_tensors[0], mask=input_masks[0]))
>   File "/directory/ve/lib/python2.7/site-packages/keras/layers/normalization.py", line 117, in call
>     epsilon=self.epsilon)
>   File "/directory/ve/lib/python2.7/site-packages/keras/backend/theano_backend.py", line 442, in normalize_batch_in_training
>     return _old_normalize_batch_in_training(x, gamma, beta, reduction_axes, epsilon)
>   File "/directory/ve/lib/python2.7/site-packages/keras/backend/theano_backend.py", line 504, in _old_normalize_batch_in_training
>     epsilon)
>   File "/directory/ve/lib/python2.7/site-packages/keras/backend/theano_backend.py", line 456, in batch_normalization
>     return _old_batch_normalization(x, mean, var, beta, gamma, epsilon)
>   File "/directory/ve/lib/python2.7/site-packages/keras/backend/theano_backend.py", line 542, in _old_batch_normalization
>     x, gamma, beta, mean, var, 'spatial', epsilon)
>   File "/directory/ve/lib/python2.7/site-packages/theano/sandbox/cuda/dnn.py", line 2857, in dnn_batch_normalization_test
>     batchnorm_op = GpuDnnBatchNormInference(mode=mode, epsilon=epsilon)
>   File "/directory/ve/lib/python2.7/site-packages/theano/sandbox/cuda/dnn.py", line 2353, in __init__
>     if version() < (5000, 5000):
>   File "/directory/ve/lib/python2.7/site-packages/theano/sandbox/cuda/__init__.py", line 413, in dnn_version
>     dnn_available.msg)
> Exception: ("We can't determine the cudnn version as it is not available", 'Device not supported')

Before adding the issue I updated Keras and Theano.

**Comment by patyork** [comment]
It looks like you may be using an old version of Theano. The Batch Normalization implementation was only available with CuDNN in old Theano versions, it appears.

**Comment by patyork** [comment]
Sorry, just saw your last line.

Please [try to clone and install this fork of Keras](https://github.com/patyork/keras/tree/issue4821), and try to run your script again.
